src/Example.Plugins.Tests/CyberArkCCPSecureStoreTests.cs

```diff
@@ -0,0 +1,284 @@
+                actual.Should().BeEquivalentTo(expected);
+
+                //Cleanup
+                store.Remove(cert);
+                store.Close();
+                store.Dispose();
+            }
+        }
+
+        [Fact]
+        public async void GetValueAsyncWithValidContextWithoutCert()
+        {
+            var actual = await customSubject.GetValueAsync(ContextIsValidTrue, "Key");
+            var expected = "Password";
+            actual.Should().BeEquivalentTo(expected);
+        }
+
+        private static HttpClientHandler CreateHttpClientHandler()
+        {
+
+            var cert = new X509Certificate2();
+            var certCollection = new X509Certificate2Collection() { cert };
+            var mock = new Mock<HttpClientHandler>();
+            mock.Object.ClientCertificates.Add(certCollection[0]);
+            
+            cert.Dispose();
+
+            return mock.Object;
+        }
+
+        private static HttpClient CreateHttpClient()
+        {
+            var handler = new CustomMsgHandler();
+            var mock = new Mock<HttpClient>(handler);
+            return mock.Object;
+        }
+
+        static X509Certificate2 CreateSelfSignedCertificateWithoutPrivateKey()
+        {
+            var ecdsa = ECDsa.Create(); // generate asymmetric key pair
+            var rsa = RSA.Create();
+            var req = new CertificateRequest("cn=foobar", rsa, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
+            var cert = req.CreateSelfSigned(DateTimeOffset.Now, DateTimeOffset.Now.AddYears(5));
+            ecdsa.Dispose();
+            rsa.Dispose();
+            return cert;
+        }
+        static X509Certificate2 CreateSelfSignedCertificateWithPrivateKey()
+        {
+            var ecdsa = ECDsa.Create(); // generate asymmetric key pair
+            var req = new CertificateRequest("cn=foobar", ecdsa, HashAlgorithmName.SHA256);
+            var cert = req.CreateSelfSigned(DateTimeOffset.Now, DateTimeOffset.Now.AddYears(5));
+            
+            // Create PFX (PKCS #12) with private key
+            File.WriteAllBytes("C:\\Temp\\CyberArkTempCert.pfx", cert.Export(X509ContentType.Pfx, "P@55w0rd"));
+            cert.Import("C:\\Temp\\CyberArkTempCert.pfx", "P@55w0rd",X509KeyStorageFlags.Exportable);
+
+            ecdsa.Dispose();
+            return cert;
+        }
+    }
+    public class CustomMsgHandler : HttpMessageHandler
+    {
+        protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            HttpContent content = new StringContent(
+                JsonConvert.SerializeObject(
+                    new CyberArkCCPPassword()
+                    {
+                        Content = "Password",
+                        Folder = "Folder",
+                        Safe = "Safe",
+                        Name = "KeyName",
+                        PasswordChangeInProcess = "No"
```

Comment:
Handled this but I was not able to fully test since I could not get the `PasswordChangeInProcess` true. I even had a while `PasswordChangeInProgress == "True"` conditional breakpoint and looped infinitely and could not hit it.